### PR TITLE
Add to upper function for strings and bytes

### PR DESCRIPTION
**Pull request by satheler**
## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

**Comment by satheler** [comment]
please review

**Comment by satheler** [comment]
please review

File: docs/usage/model_config.md [hunk 1]
```diff
@@ -28,6 +28,9 @@ _(This script is complete, it should run "as is")_
 **`anystr_lower`**
 : whether to make all characters lowercase for str & byte types (default: `False`)
 
+**`anystr_upper`**
+: whether to make all characters uppercase for str & byte types (default: `False`)
+
```
Comment (hramezani) [inline docs/usage/model_config.md:33]: As you placed `upper` on top of the lower in other places of the patch, I would suggest moving this part on top of the `anystr_lower`

File: tests/test_types.py [hunk 2]
```diff
@@ -113,6 +113,22 @@ def test_constrained_bytes_too_long():
...
+
+
+def test_constrained_bytes_upper_disabled():
+    class Model(BaseModel):
+        v: conbytes(to_upper=False)
+
+    m = Model(v=b'aBcD')
+    assert m.v == b'aBcD'
+
+
```
Comment (hramezani) [inline tests/test_types.py:131]: We can use `pytest.mark.parametrize` here:

```suggestion
@pytest.mark.parametrize(
    'to_upper, value, result',
    [
        (True, b'abcd', b'ABCD'),
        (True, b'aBcD', b'aBcD'),
    ],
)
def test_constrained_bytes_upper(to_upper, value, result):
    class Model(BaseModel):
        v: conbytes(to_upper=to_upper)

    m = Model(v=value)
    assert m.v == result
```

Also, we can apply it for other tests as well

**Comment by hramezani** [comment]
Thanks @satheler for your patch :+1: 
In general, it LGTM. I just left some comments to improve the readability.

**Comment by hramezani** [comment]
please update

**Comment by satheler** [comment]
@hramezani thank you for your suggestions. I made the suggested changes.

File: docs/usage/model_config.md (hunk 1 above)
Comment (satheler) [reply docs/usage/model_config.md:30]: adjust on 8aa6032

File: tests/test_types.py [hunk 3]
```diff
@@ -699,6 +714,22 @@ def test_constrained_str_too_long():
...
+def test_constrained_str_upper_disabled():
+    class Model(BaseModel):
+        v: constr(to_upper=False)
+
+    m = Model(v='aBcD')
+    assert m.v == 'aBcD'
+
+
```
Comment (hramezani) [inline tests/test_types.py:732]: I think we can use `pytest.mark.parametrize` for these two tests here as well

File: tests/test_types.py [hunk 4]
```diff
@@ -1810,6 +1841,37 @@ class Config:
...
+    str_check_value = 'ABCDefG'
+    bytes_check_value = b'abCD1Fg'
+
+    m = Model(str_check=str_check_value, bytes_check=bytes_check_value)
+
+    assert m.str_check == str_check_value
+    assert m.bytes_check == bytes_check_value
+
+
```
Comment (hramezani) [inline tests/test_types.py:1874]: Also, for these two tests as well, we can use `pytest.mark.parametrize

**Comment by hramezani** [comment]
Thanks @satheler for updating the patch, Please use `pytest.mark.parametrize` for other tests as well.

**Comment by hramezani** [comment]
please review

**Comment by hramezani** [comment]
please update

File: tests/test_types.py (hunk 3 above)
Comment (satheler) [reply tests/test_types.py]: refactored on ccb42d3

File: tests/test_types.py (hunk 4 above)
Comment (satheler) [reply tests/test_types.py]: refactored on 8a68576

**Comment by satheler** [comment]
@hramezani I took advantage and already did the refactoring for lower's tests.

**Comment by satheler** [comment]
`test_anystr_strip_whitespace` as well

**Comment by satheler** [comment]
please review

**Review by hramezani — APPROVED** [review]

**Comment by samuelcolvin** [comment]
thanks so much @satheler.

**Pull request merged**